src/hooks/core/useCoreChatSystemHook.ts

```diff
@@ -46,43 +56,153 @@ const useCoreChatSystemHook = () => {
     }
   };
 
-  usePostNotificationListenerHook(onPostNotifReceived);
-  const {lastJsonMessage} = useBetterWebsocketHook();
+  const handleChannelImage = (members) => {
+    if (members?.length > 2) {
+      return DEFAULT_PROFILE_PIC_PATH;
+    }
+    const findMember = members?.find((data) => data?.user_id !== signedProfileId);
+    if (findMember) {
+      return findMember?.user?.image;
+    }
+    return DEFAULT_PROFILE_PIC_PATH;
+  };
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

The `handleChannelImage` function could be optimized by removing the redundant check for `findMember`. If `findMember` is undefined, accessing its property will return `undefined`, which is a falsy value in JavaScript. So, you can directly return `findMember?.user?.image || DEFAULT_PROFILE_PIC_PATH`.
```diff
-    if (findMember) {
-      return findMember?.user?.image;
-    }
-    return DEFAULT_PROFILE_PIC_PATH;
+    return findMember?.user?.image || DEFAULT_PROFILE_PIC_PATH;
```



<!-- This is an auto-generated comment by OSS CodeRabbit -->